### PR TITLE
Make "~/.ansible" path configurable

### LINKED ISSUE #75788: Ability to easy control of ~/.ansible directory location
### Summary

I want to control ~/.ansible directory location but it's ridiculously complex. My main motivation is clean and controllable $HOME directory.

Currently I have to setup around 24 environment variables to make it possible. And even if achieve it, it is not guaranteed that behavior will not be changed in the future or new plugin will not require to fix the path.

I'm aware of previous attempts to adopt XDG:
- https://github.com/ansible/ansible/issues/52354
- https://github.com/ansible/ansible/issues/68587

I'm partially agree your points against XDG. For me personally it's not XDG_CONFIG, it should be XDG_DATA... or XDG_CACHE.

I suggest to create **one** variable which will be base prefix of rest ~**24** variables which refers to ~/.ansible directory. This variable may not have any relations with XDG at all but allow to get control to users and get rid of polluted $HOMEs.

```yaml
BASE_PREFIX:
  default: ~/.ansible
COLLECTIONS_PATHS:
  default: {{BASE_PREFIX}}/collections:/usr/share/ansible/collections
```

I offer my help to work on PR as well if you will not be against of this idea.

### Issue Type

Feature Idea

### Component Name

core

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by htol**
##### SUMMARY
I want to control ~/.ansible directory location but it's ridiculously complex. My main motivation is clean and controllable $HOME directory.

Currently I have to setup around 24 environment variables to make it possible. And even if achieve it, it is not guaranteed that behavior will not be changed in the future or new plugin will not require to fix the path.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
base.yml

Fixes ansible/ansible#75788

### CODE REVIEW

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/pull/72043

**Comment by sivel** [comment]
> Do I have any better way to collaborate?

1. Try to interact on the existing PR.  
2. Seek out interaction in #ansible-devel on IRC with the author
3. Come to an ansible core IRC meeting

**Comment by fey** [comment]
Hi.
What's left to do for pr to be accepted?

**Comment by bcoca** [comment]
change log, tests and passing existing tests

**Comment by oaken-source** [comment]
any way I can help here? I too would like this feature to make it into ansible. from what I can see from the failed tests, it seems to be mostly setup errors in the tests themselves, not any significant failures.

**Comment by htol** [comment]
Actually I have a time for continue but don't really understand next step. Can I get a clarification or direction?

**Comment by stefanwascoding** [comment]
@htol I've continued working on your PR in #77322 - mainly fixing the failing existing tests.  I would be happy for you to merge those changes back into this PR, as it seems wasteful to have two PRs for the same feature.

As far as I understand, what is missing is:
- [ ] adding documentation
- [ ] add tests for the new functionality.
- [ ] updating the change log

If you could give me write access to your fork I'd be happy to also continue working on this (on your PR).
Or, let me know if you're okay with me taking the lead on this, and I'll add you as collaborator to my fork.

**Comment by htol** [comment]
@stefanwascoding I've added you as a collaborator to repository. Let me know please if it not enough to make changes in PR. Thank you!

**Comment by stefanwascoding** [comment]
@htol Works.  Thanks.

The currently failing tests (Incidental AWS Python *) are temporary CI issues.

File: lib/ansible/plugins/loader.py
Comment (bcoca) [inline lib/ansible/plugins/loader.py:1148]: don't make this change, we eventually want to phase out the constants in favor of the function calls.

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/loader.py
Comment (sivel) [reply lib/ansible/plugins/loader.py:1148]: afaict only the constants are currently templated, `get_config_value` doesn't get templated.

Comment (stefanwascoding) [reply lib/ansible/plugins/loader.py:1148]: @bcoca as @sivel said, the `get_config_value` is not handling templates, only the constants do. See the [screenshot](https://user-images.githubusercontent.com/2292035/160599914-887296a0-a3ba-49eb-81f2-b3b0f4246f2b.png) on https://github.com/ansible/ansible/pull/77322#issuecomment-1081743289.

Personally, I would prefer to keep this change as is, and then open a separate PR, once this one is accepted, to rewrite the access of the configuration settings by providing a method (possibly by extending `get_config_value`) that is both dynamic like `get_config_value`, but also supports templating like the constants. Separate PR, because to me the modification (and code-wide usage update) of the config access seems to be too large to role into the PR about getting ANSIBLE_HOME working.

Comment (bcoca) [reply lib/ansible/plugins/loader.py:1148]: true, had meant to fix that a  while ago #77483

Comment (stefanwascoding) [reply lib/ansible/plugins/loader.py:1148]: 👍🏻  In that case, I'll wait for the PR to land in devel, rebase, and switch (back) to `get_config_value` in all places.

Comment (bcoca) [reply lib/ansible/plugins/loader.py:1148]: no need to block on that

**Review by bcoca — APPROVED** [review]

**Comment by stefanwascoding** [comment]
PR changeset currently says it [affects 2.14](https://github.com/ansible/ansible/pull/76114/files#diff-3f3fbfd37f8f95b3e5f26617feb866a00e1cddb44fad1522321927d42349b877R15), but there's an `affects_2.13` label on the PR -- might be a holdover from when the PR was created last October.

File: lib/ansible/config/base.yml
Comment (bcoca) [inline lib/ansible/config/base.yml]: I would not document as an 'env var' as it might confuse the user on how this works

**Review by bcoca — CHANGES_REQUESTED** [review]
code lgtm, just minor doc issue

File: lib/ansible/config/base.yml
Comment (htol) [reply lib/ansible/config/base.yml]: Is it a better variant?
```
``'{{ ANSIBLE_HOME ~ "/collections" }}'``
```

Comment (bcoca) [reply lib/ansible/config/base.yml]: yep

Comment (htol) [reply lib/ansible/config/base.yml]: fixed

**Review by bcoca — APPROVED** [review]

**Pull request merged**
